cluster-autoscaler/core/scaledown/unremovable/nodes.go

```diff
@@ -42,7 +42,7 @@ func NewNodes() *Nodes {
 
 // NodeInfoGetter is anything that can return NodeInfo object by name.
 type NodeInfoGetter interface {
-	Get(name string) (*schedulerframework.NodeInfo, error)
```

Comment:
Would it be safer to keep the legacy `Get` method in here given that this is an exported, public interface?

Reply:
Then we'd need to introduce an additional wrapper type. This interface is only used locally, it should be unexported IMO (it looks like the classic "local interface with a relevant subset of methods from some other interface" pattern). Changed to be unexported, WDYT?

Reply:
Right, I think this is defensible. Maybe worth calling it out in release notes.

Reply:
I think we might not be on the same page here. Why would changing an arbitrary exported interface in CA code be release-note-worthy? There's a policy around breaking changes in the `CloudProvider`/`NodeGroup` interfaces specifically, but others should be fair game to change as needed, right?

It could be argued that the `NodeGroup.TemplateNodeInfo()` signature change in this PR should be highlighted in release notes (or actually go through a deprecation?), I'm not sure about that part. All in-tree cloud providers are automatically adapted so the change doesn't break them, but it will break out-of-tree cloud providers. Not sure if that's also covered under the policy though, I think we've done similar "mechanical" changes to all in-tree cloud providers without going through the deprecation process.

@MaciekPytel WDYT?

Reply:
I'm just speaking to the fact that it's theoretically possible for an external dependancy (outside of the local CA codebase) to exist in the wild that reuses `NodeInfoGetter`, and those downstream use cases will need to be adapted to these changes.

Not a big deal either way. If that doesn't rise to the level of being included as a release note, that's fine.

Reply:
I don't think we promise to keep any sort of backward compatibility when it comes to internal interfaces in CA, besides the CloudProvider interface and processors (we did go through deprecation and multiple releases of supporting NodeInfoProcessor before dropping it). 

I'm mostly basing my opinion on kubernetes/kubernetes - I don't think there is any promise of golang interface level backward compatibility there. We certainly need to modify predicate checker pretty much every minor just to handle backward incompatible refactors in scheduler framework. So far our general practice when it comes to deprecations, supported versions and similar was to generally follow kubernetes.

All that being said - we all know there are CA forks out there. It's no secret we have one and I also assume it's no secret that so does Azure (I'm pretty sure there are issues in this repo that mention upstreaming from that fork). With processors/ CA is pretty much asking to be forked and customized. If we think this interface is somehow particularly useful for fork maintainers (e.g. based on experience of anyone in this thread) it could be a reason to leave it - perhaps with a comment discouraging new use?

Reply:
My vote would be slightly in favor of not worrying about this given the historical precedent in the k/k ecosystem of projects.

Forks are probably O.K. because they may already have implemented their own opinionated `NodeInfoGetter` anyways. :)